packages/cozy-stack-client/src/NextcloudFilesCollection.js

```diff
@@ -50,13 +57,76 @@ class NextcloudFilesCollection extends DocumentCollection {
   async download(file) {
     const res = await this.stackClient.fetch(
       'GET',
-      `/remote/nextcloud/${file.cozyMetadata.sourceAccount}/${file.path}?Dl=1`
+      `/remote/nextcloud/${file.cozyMetadata.sourceAccount}${encodePath(
+        file.path
+      )}?Dl=1`
     )
     const blob = await res.blob()
     const href = URL.createObjectURL(blob)
     const filename = file.path.split('/').pop()
     forceDownload(href, filename)
   }
+
+  /**
+   * Move a file inside a Nextcloud server
+   *
+   * @param {object} file - The file to move
+   * @param {object} to - The destination path
+   */
+  async move(file, to) {
+    const newPath = joinPath(to.path, file.name)
+    const resp = await this.stackClient.fetch(
+      'POST',
+      `/remote/nextcloud/${file.cozyMetadata.sourceAccount}/move${encodePath(
+        file.path
+      )}?To=${encodePath(newPath)}`
+    )
+    if (resp.status === 204) {
+      return resp
+    }
+    throw new FetchError(resp, resp.json())
+  }
+
+  /**
+   * Move a file from a Nextcloud server to a Cozy
+   *
+   * @param {object} file - The file to move
+   * @param {object} to - The destination folder
+   *
+   */
+  async moveToCozy(file, to) {
+    const resp = await this.stackClient.fetch(
+      'POST',
+      `/remote/nextcloud/${
+        file.cozyMetadata.sourceAccount
+      }/downstream${encodePath(file.path)}?To=${to._id}`
+    )
+    if (resp.status === 201) {
+      return resp
+    }
+    throw new FetchError(resp, resp.json())
+  }
+
+  /**
+   * Move a file from a Cozy to a Nextcloud server
+   *
+   * @param {object} file - The destination folder
```

Comment:
is it an IOCozyFiles?

Reply:
Its an Nextcloud file here. I tried to add so documentation to be more explicit in the PR (https://github.com/cozy/cozy-client/pull/1490). Unfortunately, I didn't manage to reuse the cozy-client types. Maybe we should add a 3rd package for the types to be shared.